conbencher/python/conbencher/adapters/_adapter.py

```diff
@@ -0,0 +1,86 @@
+import abc
+import subprocess
+
+from ..client import ConbenchClient
+from ..log import fatal_and_log
+from ..result import BenchmarkResult
+
+
+class _BenchmarkAdapter(abc.ABC):
+    """
+    An abstract class to run benchmarks, transform results into conbench form,
+    and send them to a conbench server
+
+    Attributes
+    ----------
+    command : list[str]
+        A list of args to be run on the command line, as would be passed
+        to `subprocess.run()`.
+    results : list[BenchmarkResult]
+        Once `run()` has been called, results from that run
+    """
+
+    command: list[str]
+    results: list[BenchmarkResult] = None
+
+    def __init__(self, command: list[str]) -> None:
+        self.command = command
+
+    def __call__(self, **kwargs) -> list:
+        """
+        Run benchmarks and post results to conbench server
+
+        Parameters
+        ----------
+        kwargs
+            Passed through to `run()`
+        """
+        self.run(**kwargs)
+        self.post_results()
+
+    def run(self, params: list[str] = None) -> list[BenchmarkResult]:
```

Comment:
If we're supporting python 3.8 (which I think we should continue) then I think this needs to be `typing.List`

Reply:
fixed this everywhere (I think)